ros2_batch_job/__main__.py

```diff
@@ -585,11 +587,18 @@ def need_package_from_pipy(pkg_name):
                 ['"%s"' % job.python, '-m', 'pip', 'uninstall', '-y'] +
                 [f'cryptography{pip_cryptography_version}', 'lxml', 'numpy'], shell=True)
         pip_cmd = ['"%s"' % job.python, '-m', 'pip', 'install', '-U']
-        if args.do_venv or sys.platform == 'win32':
-            # Force reinstall so all dependencies are in virtual environment
+        if sys.platform == 'win32':
             # On Windows since we switch between the debug and non-debug
             # interpreter all packages need to be reinstalled too
             pip_cmd.append('--force-reinstall')
+        if args.do_venv:
+            # If we are in a virtualenv, we currently expect that the vcs tool
+            # is as well.  Force install it here, even if we would have inherited
+            # it from the environment.
+            job.run(
+                ['"%s"' % job.python, '-m', 'pip', 'install', '-U', '--force-reinstall', 'vcstool'],
+                shell=True)
```

Comment:
This is seemingly a weird change to include here, but there is a reason.

During testing, I found that anytime we ran `pip install -U --force-reinstall flake8-builtins`, it would *always* upgrade `flake8` from pip, even if it could have gotten it from the operating system.  I'm honestly not sure why that is the case, but this works around it.  Since on Linux we use a new virtualenv every time, this should be safe to do.

Reply:
Ah, never mind all that.  I just found out that `pip` has a `--upgrade-strategy only-if-needed` option, which is probably what we want.